### ISSUE TITLE
viewing pictures is often impossibly slow and often does not work

### ISSUE THREAD

**Issue by cpg** [amahi/ios#43]
i often see the following behavior:

 * open a folder with a small amount of pictures
 * the first one may or may not load. the AA server does not see the requests (some times)
 * if the first one does not load (spinner on), so do not the other pictures
 * if the first one loads, often, but not always, all others are loaded in the background
 * loading images is really slow -- with a local access, it should be blazing fast. it's not unusual for a file that is < 100K to take > 5 seconds to display.

basically, it's very very flaky, it seems to fail internally and some of the algorithms for filtering may not be working well?

**Comment by csoni111** [comment]
I did some debugging and found this in the logs:
```
2018-03-12 08:50:15.174327+0530 AmahiAnywhere[5415:152118] HTTP load failing due to strict content length - expected: 470877, received: 0, received (uncompressed): 0
2018-03-12 08:50:15.174719+0530 AmahiAnywhere[5415:151773] Task <69F4A9D1-370C-44C4-8FA9-08DE5799E658>.<6> finished with error - code: -1005
```
It appears for some images randomly, where others load fine. On digging deeper I found the error description:
```
Error Domain=NSURLErrorDomain Code=-1005 "The network connection was lost." 
UserInfo={NSUnderlyingError=0x60000044eaf0 {Error Domain=kCFErrorDomainCFNetwork Code=-1005 "(null)" 
UserInfo={NSErrorPeerAddressKey=<CFData 0x604000294690 [0x1103babb0]>{length = 16, 
capacity = 16, bytes = 0x100201bbd073c9c40000000000000000}}}, 
NSErrorFailingURLStringKey=<complete url here>, NSErrorFailingURLKey=<complete url here>, 
NSLocalizedDescription=The network connection was lost.}
```

As this SO [answer](https://stackoverflow.com/a/25853478/6352482) suggests, I tried rebooting the emulator and even Reset Content & Settings, but same error appears afterwards.

This [answer](https://stackoverflow.com/a/25996971/6352482) suggests that this might happen due to ios keeping the connection open while the server closes it.

**Comment by csoni111** [comment]
@codedentwickler Yes I think LightBox is also the culprit here. It throws network request for all the files at once, instead of only loading the current, previous and next images. We can look for other image viewing libraries, but that still does not resolve `"The network connection was lost."` issue.

**Comment by csoni111** [comment]
I shall try with some different library other than LightBox and report here.

**Comment by csoni111** [comment]
This error and the one @cpg mentioned in this https://github.com/amahi/ios/issues/41#issuecomment-374482526, can be resolved now. Lightbox now [supports](https://github.com/hyperoslo/Lightbox/pull/182) option to set `preload` value which specifies how many images to preload when viewer is opened. By default it loads all at once, which make the app go out of memory if there are too many images in the directory. We can keep it to 2 (or 1 may be).
Ref. https://github.com/hyperoslo/Lightbox/blob/master/Source/LightboxConfig.swift#L42

**Comment by csoni111** [comment]
@iswiftshek can you please implement the above changes mentioned in the previous comment, so that the app does not load all the images at once.

**Comment by cpg** [comment]
When a large folder of pics is loaded, it appears that the app crashes with a memory bloating issue (according to the debugger), however, the list itself does not seem to cause trouble (though I do not see thumbnails being created on the fly). We will track this separately. Closing for now.
